core/src/main/java/tech/pegasys/web3signer/core/service/jsonrpc/handlers/signing/TransactionSerializer.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2019 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.web3signer.core.service.jsonrpc.handlers.signing;
+
+import static tech.pegasys.web3signer.core.service.jsonrpc.response.JsonRpcError.SIGNING_FROM_IS_NOT_AN_UNLOCKED_ACCOUNT;
+
+import tech.pegasys.web3signer.core.service.jsonrpc.exceptions.JsonRpcException;
+import tech.pegasys.web3signer.core.service.jsonrpc.handlers.sendtransaction.transaction.Transaction;
+import tech.pegasys.web3signer.signing.ArtifactSignerProvider;
+import tech.pegasys.web3signer.signing.SecpArtifactSignature;
+import tech.pegasys.web3signer.signing.secp256k1.Signature;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.tuweni.bytes.Bytes;
+import org.web3j.crypto.Sign.SignatureData;
+import org.web3j.crypto.TransactionEncoder;
+import org.web3j.utils.Numeric;
+
+public class TransactionSerializer {
+  private static final Logger LOG = LogManager.getLogger();
+
+  protected final ArtifactSignerProvider signer;
```

Comment:
Why did this interface change from Signer?

Reply:
Signer was an interface that ethSigner used to use from the signers library.
ArtifactSigner is the equivalent on W3S an so are the rest of the classes like ArtifactSignerProvider, ArtifactSignature.

Reply:
Even if we hadn't merged the signers library in, w3s would have had access to the Signer interface too (via the lib the same way Ethsigner does) so I'm still confused why we should change this.

Reply:
Thought I'd mentioned it because maybe we should delete that Signer interface...
but on closer inspection it looks like ArtifactSigner might be wrapping that Signer interface. 
Maybe not for this PR, but feels like might be a refactoring opportunity here.